src/gs-updates-page.c

```diff
@@ -919,6 +943,8 @@ _add_app_row (GsUpdatesPage *self, GsApp *app)
 		_create_listbox_section (self, section);
 
 	app_row = gs_app_row_new (app);
+	g_object_set_data_full (G_OBJECT (app_row), "page", g_object_ref (self),
+				g_object_unref);
```

Comment:
I don't think the ref/unref are required here, since the app row is contained within the page hierarchy; when it's destroyed the app row will be destroyed too.